src/libraries/System.Security.Cryptography.Xml/tests/SignedXmlTest.cs

```diff
@@ -1677,36 +1676,80 @@ public void VerifyXmlResolver(bool provideResolver)
-
-                ((IDisposable)listener).Dispose();
             }
 
             static async Task ProcessRequests(
-                HttpListener listener,
-                Action<HttpListenerRequest> requestReceived,
+                TcpListener listener,
+                Action requestReceived,
                 CancellationToken cancellationToken)
             {
+                const string Response =
+                    "HTTP/1.1 200 OK\r\nContent-Type: text/plain; charset=UTF-8\r\nContent-Length: 0\r\n\r\n";
+
                 while (!cancellationToken.IsCancellationRequested)
                 {
-                    HttpListenerContext ctx;
+                    Socket socket;
 
                     try
                     {
-                        ctx = await listener.GetContextAsync();
+#if NETCOREAPP
+                        socket = await listener.AcceptSocketAsync(cancellationToken);
+#else
+                        socket = await listener.AcceptSocketAsync();
+#endif
                     }
                     catch
                     {
                         break;
                     }
 
-                    HttpListenerRequest req = ctx.Request;
-                    requestReceived(req);
-
-                    using (HttpListenerResponse resp = ctx.Response)
+                    using (socket)
                     {
-                        resp.ContentType = "text/plain";
-                        resp.ContentEncoding = Encoding.UTF8;
-                        resp.ContentLength64 = 0;
+                        requestReceived();
+                        byte[] buf = new byte[1024];
+
+#if NETCOREAPP
+                        // The CoreFX XML libraries throw an exception if we don't drain the request
+                        // (unless we don't dispose the socket).
+                        // For some reason each ReceiveAsync only gets 10 bytes at a time,
+                        // so just keep reading.
+                        int offset = 0;
+
+                        do
+                        {
+                            int read = await socket.ReceiveAsync(buf.AsMemory(offset), cancellationToken);
+
+                            if (read <= 0)
+                            {
+                                break;
+                            }
+
+                            offset += read;
+
+                            if (offset >= buf.Length)
+                            {
+                                throw new InvalidOperationException();
+                            }
+
+                            if (offset > sizeof(int))
+                            {
+                                int last = BinaryPrimitives.ReadInt32LittleEndian(buf.AsSpan(offset - sizeof(int)));
+
+                                // CRLFCRLF, little-endian
+                                if (last == 0x0A0D0A0D)
+                                {
+                                    break;
+                                }
```

Comment:
Could we simplify to just:
```C#
if (buf.AsSpan(offset - sizeof(int)).SequenceEqual("\r\n\r\n"u8))
{
    break;
}
```
? It'd be easier to read, and even if it's a tad slower, this is a test :) Plus, I _thought_ this case would be recognized by the JIT and unrolled, but that doesn't appear to be happening (cc: @EgorBo), so I may be misremembering the exact types/arguments supported for that (maybe byte support never made it in?)